free-2.md

```diff
@@ -1,69 +1,168 @@
 # Free project 2
 
 
+
+This project would be something like the functional equivalent of [scratch][scratch.mit.edu], only as a learning tool for textual coding. It would have a split-screen interface with the area for writing code on the left (maybe in a new language, but most likely in a pre-existing one), and on the right would be an image visualiation of the code, with similar color-coded "building blocks" to show exactly what functions are being called where and what's being passed into them. As code is added, there could be a button to "refresh" the visualization area, which would also be updated when the code is executed.
+
+
+
 ## The user and a language
+
 This section describes who the project would serve and why a language might be a
+
 good way to meet their needs.
 
 
+
+This project would be for people who are trying to learn functional programming, either for their first exposure to CS or for someone who's used to imperative programming.
+
+
+
 ### What's the need?
+
 _What need is met by your idea? Who are you helping? What is that person's
+
 experience like now? What would their experience be like if you could help 
+
 them?_
 
 
+
+Imperative programming seems to be more commonly used for entry-level compter programmers, and while it has its benifits this makes people more hesitant or have more difficulty with undertstanding functional programming, which in some cases can be a more effective tool for the problem they're trying to solve. In addition, some people are more visual learners, and having a view of what's happening in a program would make debugging easier. With a learning tool, picking up a more foreign language would be easier and faster, and new programmers might start out with a deeper understanding of computer programming.
```

Comment:
I agree that it would be good to have a functional on-ramp to programming. I though CS 5 did a really good job of emphasizing functional programming.